sqrl-planner/src/main/java/com/datasqrl/plan/rules/SQRLLogicalPlanRewriter.java

```diff
@@ -231,21 +245,26 @@ public RelNode visit(LogicalValues logicalValues) {
 
   private PrimaryKeyMap determinePK(LogicalValues logicalValues) {
     ImmutableList<ImmutableList<RexLiteral>> tuples = logicalValues.getTuples();
-    if (tuples.size()<2) return PrimaryKeyMap.none();
-    //Iterate over all columns and pick the first one with unique values
-    int numRows = tuples.size();
-    List<RelDataTypeField> fields = logicalValues.getRowType().getFieldList();
-    for (int i = 0; i < tuples.get(0).size(); i++) {
-      int colNo = i;
-      RelDataType type = fields.get(colNo).getType();
+    if (tuples.size()<=1) return PrimaryKeyMap.none();
+    return determinePK(logicalValues.getRowType());
+  }
+
+  /**
+   * Determines the primary key of a table by rowtype using the simple heuristic of selecting
+   * the first non-null scalar column.
+   *
+   * @param rowType
+   * @return
+   */
+  private PrimaryKeyMap determinePK(RelDataType rowType) {
+    List<RelDataTypeField> fields = rowType.getFieldList();
+    for (int i = 0; i < fields.size(); i++) {
+      RelDataType type = fields.get(i).getType();
       if (!CalciteUtil.isPotentialPrimaryKeyType(type)) continue;
-      long uniques = tuples.stream().map(t -> t.get(colNo).getValue()).distinct().count();
-      if (numRows == uniques) {
-        //This is a primary key
-        return PrimaryKeyMap.of(List.of(colNo));
-      }
+      return PrimaryKeyMap.of(List.of(i));
```

Comment:
Let's make sure this is documented since it can be surprising behavior

Reply:
Across the board, we now assume that the first not-null scalar column for nested data is the primary key. If that's wrong, there is a hint the user can use to overwrite. This needs to be added to the docs.